### PR TITLE
Added helper functions for facility xwalks

### PR DESCRIPTION
**Pull request by erika-tyagi**
Two more quick helper functions, useful in the short-term for facility crosswalks, potentially for other workflows too? 

Right now, the only valid state spellings are the 50 states, DC, Puerto Rico, Guam, Virgin Islands, and Not Available. As a result, invalid spellings (intentionally) include Federal, District of Columbia, and Abroad. Is this too strict? Is there anything else I should include as valid? 

Also, the current crosswalk includes Not Available, NA, Abroad, DC and District of Columbia listed separately, Deleware and Delaware listed separately, etc. I can clean this up when I reset the IDs - unless it would break things? 

Opening PRs that grant statehood to territories is a lot of pressure.

### CODE REVIEW

**Comment by nmmarquez** [comment]
Hey I think these are useful functions, is it possible vectorize these functions so that they work with more than 1 HIFLD or state at a time?

right now I get weird warnings and variable output when passing a vector of id values or states

**Comment by hjohns12** [comment]
It looks like there is a prison in American Samoa called Tafuna Correctional Facility (pretty sure we're not collecting data about it though). Right now, I think this would get categorized under "Not available". To be consistent, I think we should add in the other US territories to the list of possible states. 

Also just discovered that there is a prison in the Northern Mariana Islands, although I couldn't find any COVID data online.

File: R/is_valid_state.R [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#' is_valid_state("DC")
+#' is_valid_state("District of Columbia")
+
+is_valid_state <- function(state, na_is_na = T) {
+    
+    if (is.na(state) & na_is_na) {return(NA)}
+    
+    valid_states <- datasets::state.name %>% 
+        append(c(
+            "DC",
```
Comment (nmmarquez) [inline R/is_valid_state.R:26]: I think we are changing DC to District of Columbia in the new info sheet

**Comment by erika-tyagi** [comment]
Thanks! Wholehearted agree with all comments. Updates:  
* Vectorized the functions (and added examples of passing multiple IDs/states in) 
* Added all 5 of the populated territories 
* Changed the default `na_is_na` to F. The main reason I initially wrote them to return T/F/NA instead of T/F is because I wanted to handle cases where a user didn't list an HIFLD (NA) differently from a case where they listed an invalid HIFLD (typo) in the Google Sheet - but I'm open to removing that option from these functions! 
* Changed DC to District of Columbia

**Comment by hjohns12** [comment]
Looks ready to merge to me!

**Comment by nmmarquez** [comment]
So im a big dummy and misinterpreted the `na_is_na` paramter. I totally think it should be `TRUE` after testing. @erika-tyagi can we 1) switch it back, and 2) please forgive me

**Comment by erika-tyagi** [comment]
> So im a big dummy and misinterpreted the `na_is_na` paramter. I totally think it should be `TRUE` after testing. @erika-tyagi can we 1) switch it back, and 2) please forgive me

@nmmarquez Done! Renamed to `missing_as_na` and defaults to TRUE. You're the opposite of a dummy, and very much forgiven!

**Pull request merged**